build-system/pr-check/e2e-tests.js

```diff
@@ -21,31 +21,49 @@
  * This is run during the CI stage = test; job = e2e tests.
  */
 
+const colors = require('ansi-colors');
 const {
   downloadBuildOutput,
   downloadDistOutput,
   printChangeSummary,
   startTimer,
   stopTimer,
   timedExecOrDie: timedExecOrDieBase} = require('./utils');
+const {determineBuildTargets} = require('./build-targets');
 const {isTravisPullRequestBuild} = require('../travis');
 
 const FILENAME = 'e2e-tests.js';
+const FILELOGPREFIX = colors.bold(colors.yellow(`${FILENAME}:`));
 const timedExecOrDie =
   (cmd, unusedFileName) => timedExecOrDieBase(cmd, FILENAME);
 
 async function main() {
   const startTime = startTimer(FILENAME, FILENAME);
+  const buildTargets = determineBuildTargets();
 
   if (!isTravisPullRequestBuild()) {
     downloadDistOutput(FILENAME);
+    timedExecOrDie('gulp update-packages');
+    timedExecOrDie('gulp e2e --nobuild --headless');
   } else {
     printChangeSummary(FILENAME);
-    downloadBuildOutput(FILENAME);
-  }
-  timedExecOrDie('gulp update-packages');
-  timedExecOrDie('gulp e2e --nobuild --headless');
+    if (buildTargets.has('RUNTIME') ||
+        buildTargets.has('UNIT_TEST') ||
+        buildTargets.has('INTEGRATION_TEST') ||
+        buildTargets.has('BUILD_SYSTEM') ||
+        buildTargets.has('FLAG_CONFIG') ||
+        buildTargets.has('VISUAL_DIFF')) {
```

Comment:
Why are these relevant to visual diffs?

Reply:
Not exactly relevant, but I'd like to run this job whenever possible, which means whenever a build artifact is created. By doing so we'd catch errors more quickly if something external changed, like browser versions.